swarm/api/client/client.go

```diff
@@ -562,3 +563,111 @@ func (c *Client) MultipartUpload(hash string, uploader Uploader) (string, error)
 	}
 	return string(data), nil
 }
+
+// CreateResource creates a Mutable Resource with the given name and frequency, initializing it with the provided
+// data. Data is interpreted as multihash or not depending on the multihash parameter.
+// startTime=0 means "now"
+// Returns the resulting Mutable Resource manifest address that you can use to include in an ENS Resolver (setContent)
+// or reference future updates (Client.UpdateResource)
+func (c *Client) CreateResource(name string, frequency, startTime uint64, data []byte, multihash bool, signer mru.Signer) (string, error) {
```

Comment:
I totally agree this is what we should do. However, if you look at the other methods in `client.go`, you will see that most don't expose internal structures and simplify things to the user by dealing with hex strings instead of a hash type, for example.

If it is OK, I'd really like to just expose the structures and avoid so many parameters.

Reply:
Well if we separate the updates out that two parameters less so it's fine.